validator/sawtooth_validator/networking/interconnect.py

```diff
@@ -253,7 +276,7 @@ class Interconnect(object):
     def __init__(self,
                  endpoint,
                  dispatcher,
-                 identity=None,
+                 zmq_identity=None,
```

Comment:
Yes, they are different. This is the zmq_identity which represents incoming connections. The connection_id is either a hash of this identity or a hash of the connection string for an OutboundConnection object.